src/main/java/hudson/plugins/s3/S3BucketPublisher.java

```diff
@@ -187,9 +187,15 @@ public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath ws, @Nonnull Launc
                 if (paths.isEmpty()) {
                     // try to do error diagnostics
                     log(listener.getLogger(), "No file(s) found: " + expanded);
```

Comment:
Please, extract this code branch to new private method - something like 'printDiagnostics'.